src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -32,12 +33,20 @@ class ExpressionLanguage
     protected $functions = array();
 
     /**
-     * @param ParserCacheInterface                  $cache
+     * @param CacheItemPoolInterface                $cache
      * @param ExpressionFunctionProviderInterface[] $providers
      */
-    public function __construct(ParserCacheInterface $cache = null, array $providers = array())
+    public function __construct($cache = null, array $providers = array())
     {
-        $this->cache = $cache ?: new ArrayParserCache();
+        if (null !== $cache && false === is_a($cache, CacheItemPoolInterface::class)) {
```

Comment:
Should be:
```php
if (null !== $cache && !$cache instanceof CacheItemPoolInterface) {
```

Since 5.0.0 :P
> This function became deprecated in favour of the instanceof operator. Calling this function will result in an E_STRICT warning.